src/connector/src/parser/common.rs

```diff
@@ -42,16 +51,43 @@ fn do_parse_simd_json_value(dtype: &DataType, v: &BorrowedValue<'_>) -> Result<S
                 .map_err(|e| anyhow!("expect i32: {}", e))?,
         ),
         DataType::Int64 => ensure_int!(v, i64).into(),
-        DataType::Float32 => ScalarImpl::Float32((simd_json_ensure_float!(v, f32) as f32).into()),
-        DataType::Float64 => ScalarImpl::Float64((simd_json_ensure_float!(v, f64)).into()),
+        // when f32 overflows, the value is converted to `inf` which is inappropriate
+        DataType::Float32 => {
+            let scalar_val = ScalarImpl::Float32((simd_json_ensure_float!(v, f32) as f32).into());
+            if let ScalarImpl::Float32(f) = scalar_val {
+                if f.is_infinite() {
+                    anyhow::bail!("{v} is out of range for type f32");
+                }
+            }
+            scalar_val
+        }
+        DataType::Float64 => {
+            let scalar_val = ScalarImpl::Float64((simd_json_ensure_float!(v, f64)).into());
+            if let ScalarImpl::Float64(f) = scalar_val {
+                if f.is_infinite() {
+                    anyhow::bail!("{v} is out of range for type f64");
+                }
+            }
+            scalar_val
+        }
         // FIXME: decimal should have more precision than f64
         DataType::Decimal => Decimal::from_f64(simd_json_ensure_float!(v, Decimal))
             .ok_or_else(|| anyhow!("expect decimal"))?
             .into(),
         DataType::Varchar => ensure_str!(v, "varchar").to_string().into(),
         DataType::Bytea => ensure_str!(v, "bytea").to_string().into(),
-        DataType::Date => str_to_date(ensure_str!(v, "date"))?.into(),
-        DataType::Time => str_to_time(ensure_str!(v, "time"))?.into(),
+        // debezium converts date to i32 for mysql and postgres
+        DataType::Date => match v {
+            BorrowedValue::String(_) => str_to_date(ensure_str!(v, "date"))?.into(),
```

Comment:
Same as the bool case above, you are already guaranteed a string and there is no need to `ensure` again:
```
BorrowedValue::String(s) => str_to_date(s)?.into()
```